uvicorn/supervisors/multiprocess.py

```diff
@@ -63,8 +63,13 @@ def startup(self) -> None:
             self.processes.append(process)
 
     def shutdown(self) -> None:
+        for sock in self.sockets:
+            sock.close()
+
```

Comment:
Why do you need to close the sockets here again?

Reply:
In UNIX-based OSes one needs to close all instances oof listening socket for it to unbind.  In this case this includes:
* master process.
* all childen.

If you're hinting that in some OS'es (e.g. Windows, where my knowledge is quite limited) this would not be needed I can wrap this in: `try/except socket.error`.

Reply:
You can test this behaviour with a server that has a health check handler and a a shutdown lifespan:
```
    @app.on_event("shutdown")
    async def shutdown():
        await asyncio.sleep(60)
```